src/parser.py

```diff
@@ -0,0 +1,56 @@
+from pathlib import Path
+from parsed_codeblock import ParsedCodeBlock
+
+class CodeblockParser:
+    def parse_text(self, text: str, path: Path) -> list:
+        parsed_codeblocks = []
+        current_codeblock: ParsedCodeBlock = None
+        
+        for (line_number, line_raw) in enumerate(text.splitlines()):
+            # FIXME codeblocks usually have offset
+            # trim that off for neater looking anki notes
+            # line = line_raw.rstrip().lstrip()
```

Comment:
this seems like more effort than its  worth, delete this comment, no need to address that